src/js/with-react.jsx

```diff
@@ -0,0 +1,95 @@
+import dasherize from './dasherize';
+import on from './on';
+
+/**
+ * Provides a partially applied function which let's you wrap any WebComponent with React.
+ * - it supports first-class props for web components
+ * - it handles custom events
+ *
+ * @link https://github.com/webcomponents/react-integration - inspired by react-integration
+ * @param React
+ * @param {Boolean} [options.pure=true] - Is this a pure component?
+ * @returns {function(*)}
+ *
+ * @example <caption>How to use</caption>
+ * import React from 'react';
+ * import withReact from '@axa-ch/patterns-library/src/js/with-react';
+ * import AXAButton from '@axa-ch/patterns-library/dist/components/m-button'
+ *
+ * const AXAButtonReact = withReact(React)(AXAButton);
+ *
+ * const MyApp = ({ color, onClick }) => (
+ *  <AXAButtonReact color={color} onClick={onClick}>Hello World</AXAButtonReact>
+ * );
+ */
+const withReact = (React, { pure = true } = {}) => (WebComponent) => {
+  const { name } = WebComponent;
+  const displayName = `${name}React`;
+  const WCTagName = dasherize(WebComponent.name);
+  const eventCache = {};
+  const Component = pure ? React.PureComponent : React.Component;
+
+  return class WebComponentWrapper extends Component {
+    static get displayName() {
+      return displayName;
+    }
+
+    constructor(props) {
+      super(props);
+
+      this.handleRef = this.handleRef.bind(this);
+    }
+
+    componentDidMount() {
+      this.componentWillReceiveProps(this.props);
+    }
+
+    componentWillReceiveProps(props) {
+      const { wcNode } = this;
+
+      Object.keys(props).forEach((key) => {
+        if (key === 'children' || key === 'style') {
```

Comment:
better an external enum with an array of properties

Reply:
and also maybe use this on the property setting on the WC abstract component

Reply:
Do I understand you correctly, that I should put `'children'` and `'style'` into a blacklist array? If so absolutely

Reply:
done